src/url-handling.js

```diff
@@ -39,37 +41,49 @@ function dataSourceDirs(urlState: UrlState) {
 
 // "null | void" in the query objects are flags which map to true for null, and false
 // for void. False flags do not show up the URL.
-type BaseQuery = {
+type BaseQuery = {|
   range?: string, //
   thread?: string, // "3"
-  threadOrder?: string, // "3-2-0-1"
-  hiddenThreads?: string | void, // "0-1"
+  globalTrackOrder?: string, // "3-2-0-1"
+  hiddenGlobalTracks?: string, // "0-1"
+  hiddenLocalTracksByPid?: string,
+  localTrackOrderByPid?: string,
+  file?: string, // Path into a zip file.
   react_perf?: null, // Flag to activate react's UserTimings profiler.
   transforms?: string,
-};
+|};
 
-type CallTreeQuery = BaseQuery & {
+type CallTreeQuery = {|
+  ...BaseQuery,
   search?: string, // "js::RunScript"
   invertCallstack?: null | void,
   implementation?: string,
-};
+|};
 
-type MarkersQuery = BaseQuery & {
+type MarkersQuery = {|
+  ...BaseQuery,
   markerSearch?: string, // "DOMEvent"
-};
+|};
 
-type StackChartQuery = BaseQuery & {
+type StackChartQuery = {|
+  ...BaseQuery,
   search?: string, // "js::RunScript"
   invertCallstack?: null | void,
   implementation?: string,
+|};
+
+// Be very permissive in how the Query object is defined. Otherwise we would have to
```

Comment:
you mean "permissive" as in "not exact" ?

Reply:
I rewrote the comment:

```
// Use object type spread in the definition of Query rather than unions, so that they
// are really easy to manipulate. This permissive definition makes it easy to not have
// to refine the type down to the individual query types when working with them.
```